engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeEncoders.scala

```diff
@@ -21,12 +21,15 @@ object TypeEncoders {
   private val encodeUnknown = JsonObject("refClazzName" -> fromString(classOf[Object].getName),
     "params" -> fromValues(Nil))
 
-  private def encodeTypingResult(result: TypingResult): Json = fromJsonObject((result match {
-    case single: SingleTypingResult => encodeSingleTypingResult(single)
-    case typing.Unknown => encodeUnknown
-    case TypedUnion(classes) =>
-      JsonObject("union" -> fromValues(classes.map(encodeTypingResult).toList))
-  }).+:(typeField -> fromString(TypingType.forType(result).toString)))
+  private def encodeTypingResult(result: TypingResult): Json = fromJsonObject(
+    (result match {
+      case single: SingleTypingResult => encodeSingleTypingResult(single)
+      case typing.Unknown => encodeUnknown
+      case TypedUnion(classes) =>
+        JsonObject("union" -> fromValues(classes.map(encodeTypingResult).toList))
+    })
+      .+:(typeField -> fromString(TypingType.forType(result).toString))
+      .+:("display" -> fromString(result.display)))
```

Comment:
@arkadius `display` is a pretty version of ref.klass.getName; this is, instead of `java.lang.Long` `Long` is displayed; instead of `java.lang.Object` `Unknown` is displayed.
We could have independent implementation, only for FE on FE side, but in some cases it would be tricky to implement on FE.
eg. for `TypedClass#display` currently we're using Java reflection:
```
    override def display: String = {
      val className = ReflectUtils.fixedClassSimpleNameWithoutParentModule(ClassUtils.primitiveToWrapper(klass))
      if (params.nonEmpty)
        s"$className[${params.map(_.display).mkString(",")}]"
      else
        s"$className"
    }
```

But I see your point, that it would be nice to separate FE and BE resposibilities, but maybe for now using TypingResult#display is acceptable, and we address the issue in a separate PR.?

Reply:
How types of parameters are formatted now on FE?
I wonder if current implementation of displaying types in parameters is good enough to use for inferred types or just need to have some little tweaks to be fair enough? I don't want to make both implementations go far away from themself because at the end IMO we would like to have the same representation in both places. WDYT?

Reply:
Ah, ok I see, indeed we have the same functionality implemented differently for parameters:
// NodeDetailsContent.js
```
  renderFieldLabel = (label) => {
    const parameter = this.findParamDefinitionByName(label)
    return (
      <div className="node-label" title={label}>{label}:
        {parameter ?
          <div className="labelFooter">{ProcessUtils.humanReadableType(parameter.typ.refClazzName)}</div> : null}
      </div>
    )
  }
```

I agree, it should be in one place. I'll change that.

Reply:
@arkadius

Reply:
@arkadius 
I've changed how type labels are displayed - now we're using information from BE, instead of `ProcessUtils.humanReadableType` in all places.
https://github.com/TouK/nussknacker/pull/1191/commits/ffab8ab0bcebe973411ff5c60e68e23e65a74bf3

!but, this changes how types are displayed:
Object -> Unknown
List -> List[Unknown]
List -> List[Long]

Reply:
1. Now it is concise and it is fine
2. New types doesn't look too much verbose so it is IMO also fine
3. IMO `humanReadableType` method should stay. I would change it to `typeLabel` or sth like that and pass to it whole type info (not only value of display). Thanks to that we could change the implementation in one place. Maybe in the future we could create component rendering this type...

Reply:
@arkadius 
Ad.3 The same thinkgwas suggested by @mproch 
I've restored #humanReadableType
https://github.com/TouK/nussknacker/pull/1191/commits/102900b3a62dfa10743ba361b0d6d3f6c5cf61ac

Reply:
Indeed! :+1: